### PR TITLE
ARTEMIS-1230 Added artemis-bom

### PR DESCRIPTION
**Pull request by amarkevich**
Rework https://github.com/apache/activemq-artemis/pull/2615
- excluded test dependencies;
- introduce parent submodule

### CODE REVIEW

**Comment by amarkevich** [comment]
> @amarkevich, this needs to be rebased since there are conflicts now.

done

File: artemis-parent/pom.xml
Comment (gemmellr) [inline artemis-parent/pom.xml:26]: Something else to consider might be whether we instead introduced a new name at the root pom (e.g artemis-project, artemis-parent...could still use e.g build-parent here later) and leave this as artemis-pom for now, perhaps avoiding the change of effective content breaking anyone...e.g some might be [ab]using it as an import due to lack of specific bom.

(or also like it breaks the now-standalone Examples build, which currently still uses it as a parent as it always had...in part as that was the minimal change during their recent move, but also that there was no bom yet and I didnt want to import the parent during the move or duplicate all the management).

File: artemis-parent/pom.xml
Comment (amarkevich) [reply artemis-parent/pom.xml]: also changed property to `-Djava.library.path="${activemq.basedir}/target/bin"`

File: artemis-parent/pom.xml
Comment (amarkevich) [reply artemis-parent/pom.xml]: in general rest of transitive dependencies are here but not a top level one

File: artemis-parent/pom.xml
Comment (gemmellr) [reply artemis-parent/pom.xml]: I'm not sure changing that is correct, the files will go in subdirs which is what the property contents pointed to I believe. It was just that you moved the extraction config and so changed the effect of where they were being extracted to that was the issue.

File: artemis-parent/pom.xml
Comment (gemmellr) [reply artemis-parent/pom.xml]: right, I'm saying we should decide what we want to do with them...though I guess we can do that later.

File: artemis-pom/pom.xml
Comment (gemmellr) [inline artemis-pom/pom.xml:1174]: Seems like an aggregrate goal usage should be be in the root?

File: artemis-pom/pom.xml
Comment (gemmellr) [inline artemis-pom/pom.xml:1259]: Feels like the enforcement should be at the root at least to fail before doing anything.

File: artemis-pom/pom.xml
Comment (gemmellr) [inline artemis-pom/pom.xml:1279]: This needs to be at the root to continue having the desired effect, since it was to suppress activity at the root.

File: artemis-pom/pom.xml
Comment (gemmellr) [inline artemis-pom/pom.xml:1778]: Feels like this one should be at the root also

File: artemis-pom/pom.xml
Comment (gemmellr) [inline artemis-pom/pom.xml:1816]: Feels like these should be at the root as they are overrides

File: artemis-pom/pom.xml
Comment (gemmellr) [inline artemis-pom/pom.xml:2046]: another aggregate, does it need to be at root?

File: artemis-bom/pom.xml
Comment (gemmellr) [inline artemis-bom/pom.xml:1]: This new file is seemingly using Windows line endings according to the git diff, can you convert it to Unix line endings to match the rest.

File: artemis-pom/pom.xml
Comment (gemmellr) [inline artemis-pom/pom.xml:1]: Similarly, this new file has Windows line endings, switch it to Unix for consistency with others.

**Review by gemmellr — APPROVED** [review]
LGTM. Verified with a full test suite run and verified the examples against it.

I'll give others some more time to review first before I merge.

**Comment by jbertram** [comment]
LGTM.

**Comment by tabish121** [comment]
Changes look okay to me

**Pull request merged**
